spec/functions/number.md

```diff
@@ -657,6 +657,14 @@ or any option-specific lower limit, a _Bad Option Error_ is emitted.
 
 ### Number Selection
 
+The value of the `select` _option_ MUST be set by a _literal_,
+as otherwise the message might not be translatable.
+If this value is set by a _variable_ or
+the option value of an implementation-defined type used as an _operand_,
+a _Bad Option Error_ is emitted and
+the _resolved value_ of the expression MUST NOT support selection.
```

Comment:
link cleanup

```suggestion
as otherwise the _message_ might not be translatable.
If this value is set by a _variable_ or
the _option_ _value_ of an implementation-defined type used as an _operand_,
a _Bad Option Error_ is emitted and
the _resolved value_ of the expression MUST NOT support selection.
```

Reply:
The first fix is valid, but _option_ _value_ is incorrect when referring to an implementation-defined type, because its values are not syntax values.
```suggestion
as otherwise the _message_ might not be translatable.
If this value is set by a _variable_ or
the option value of an implementation-defined type used as an _operand_,
a _Bad Option Error_ is emitted and
the _resolved value_ of the expression MUST NOT support selection.
```